modules/bmc/redfish.rb

```diff
@@ -0,0 +1,210 @@
+require 'redfish_client'
+require 'bmc/base'
+require 'bmc/redfish/dell'
+require 'bmc/redfish/hpe'
+
+module Proxy
+  module BMC
+    class Redfish < Base
+      include Proxy::Log
+      include Proxy::Util
+
+      def initialize(args)
+        @redfish_verify_ssl = Proxy::BMC::Plugin.settings.bmc_redfish_verify_ssl
+        # enforce SSL verification as a default, if the value isn't set in the config
+        # file, or is set to a "garbage" value (anything but boolean false)
+        @redfish_verify_ssl = true unless @redfish_verify_ssl == false
+        super
+        load_vendor_overrides
+        @bmc
+      end
+
+      def connect(args = { })
+        # TODO probably verify should be an option.
+        connection = RedfishClient.new("https://#{args[:host]}/", verify: @redfish_verify_ssl)
+        connection.login(args[:username], args[:password])
+        connection
+      end
+
+      def load_vendor_overrides
```

Comment:
Definitely, this is very hacky :-)

Reply:
I was thinking about the fact that there are many implementations of redfish, with many variations, and I could only test a small number of them, and it seemed like it might be valuable to give folks a way to provide their own overrides, either to maintain locally, or as a first step toward contributing them, without necessarily needing to hack up the "official" part of the code. If you don't think it's valuable, or prefer something else, I'll change it.

Reply:
Oops, I ran out of time to edit - the overrides may not look like much now, but that has a lot to do with luck ultimately deriving from the limited scope of what's implemented today in the proxy (I only implemented what the IPMI provider currently offers, though Redfish could realistically provide much more). If the feature list starts to grow beyond what's already here, even a little, there starts to be much more divergence between vendor implementations.

Reply:
I guess since I have to require the vendor extensions anyway, "without necessarily needing to hack up the official part of the code" isn't a coherent idea. I'll think a little harder about it.

Reply:
The point is, instead of playing with strings and replacing/tring etc you could have just a simple map which is safer. Granted if the implementation adds a new vendor the map would probably need an update too but in regard to readability and sustainability it's probably better.